Services/Administration/OJS.Services.Administration.Business/Implementations/ProblemsBusinessService.cs

```diff
@@ -127,14 +127,14 @@ public async Task<ServiceResult> CopyToContestByIdByContestAndProblemGroup(int i
 
         public async Task<bool> UserHasProblemPermissions(int problemId, string? userId, bool isUserAdmin)
         {
-            var problem = await this.problemsData.OneByIdTo<ProblemShortDetailsServiceModel>(problemId);
+            var problem = await this.problemsData.OneById(problemId);
 
             if (problem == null)
             {
                 throw new BusinessServiceException("Problem cannot be null");
             }
 
-            return await this.contestsBusiness.UserHasContestPermissions(problem.ContestId, userId, isUserAdmin);
+            return await this.contestsBusiness.UserHasContestPermissions(problem.ProblemGroup.Contest.Id, userId, isUserAdmin);
```

Comment:
Why this is changed? Don't we have `problem.ContestId` anymore? What happens with `ProblemShortDetailsServiceModel`?
Also `problem.ProblemGroup.ContestId` is better than `problem.ProblemGroup.Contest.Id`.

Reply:
Reverted changes